### PR TITLE
DOCSP-25528 cleanup

### PR DESCRIPTION
**Pull request by jordan-smith721**
# Pull Request Info

- Cleaned up TODOs
- Added Read Operations to TOC
- [Added Usage Examples landing page](https://docs-mongodbcom-staging.corp.mongodb.com/csharp/docsworker-xlarge/DOCSP-25528-cleanup/usage-examples/)
- [Added Fundamentals landing page](https://docs-mongodbcom-staging.corp.mongodb.com/csharp/docsworker-xlarge/DOCSP-25528-cleanup/fundamentals/#std-label-csharp-fundamentals)
- [Added driver overview page](https://docs-mongodbcom-staging.corp.mongodb.com/csharp/docsworker-xlarge/DOCSP-25528-cleanup/)

[PR Reviewing Guidelines](https://github.com/mongodb/docs-golang/blob/master/REVIEWING.md)

JIRA - https://jira.mongodb.org/browse/DOCSP-25528
Staging - https://docs-mongodbcom-staging.corp.mongodb.com/csharp/docsworker-xlarge/DOCSP-25528-cleanup/

## Self-Review Checklist

- [] Is this free of any warnings or errors in the RST? <- One ref won't work until a separate PR is merged in
- [x] Did you run a spell-check?
- [x] Did you run a grammar-check?
- [] Are all the links working? <- One ref won't work until a separate PR is merged in

### CODE REVIEW

File: source/fundamentals/crud.txt [hunk 1]
```diff
@@ -9,4 +9,5 @@ CRUD Operations
 .. toctree::
    :caption: CRUD Operations
 
-   /fundamentals/crud/write-operations
\ No newline at end of file
+   /fundamentals/crud/write-operations
+   /fundamentals/crud/read-operations
```
Comment (rustagir) [inline source/fundamentals/crud.txt:13]: Issue: these sections don't have navigable landing pages from the TOC but you can navigate to them from the arrows that  bring you through all of the pages.

Suggestion: at least add a toc to these pages with links to the drawer pages.

File: source/fundamentals/crud/read-operations.txt [hunk 2]
```diff
@@ -0,0 +1,15 @@
+.. _csharp-crud-write-operations:
```
Comment (rustagir) [inline source/fundamentals/crud/read-operations.txt:1]: Issue: this tag says write operations

File: source/fundamentals/crud/read-operations.txt [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+
+===============
+Read Operations
+===============
+
+.. default-domain:: mongodb
+
+- :ref:`csharp-delete-guide`
```
Comment (rustagir) [inline source/fundamentals/crud/read-operations.txt:9]: Issue: the TOC here is not accurate to the pages actually contained under Read Operations

File: source/fundamentals/crud/read-operations.txt [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+
+.. toctree::
+   :caption: Write Operations
```
Comment (rustagir) [inline source/fundamentals/crud/read-operations.txt:12]: Issue: this caption says write operations

File: source/index.txt [hunk 5]
```diff
@@ -13,5 +13,59 @@ MongoDB C# Driver
...
    /issues-and-help
    /faq
-   
 
+Introduction
+------------
+
+Welcome to the documentation site for the official {+driver-long+}. 
+You can add the driver to your application to work with MongoDB in {+language+}. 
+Download it using `NuGet <https://www.nuget.org/>`__, or set up a runnable
```
Comment (rustagir) [inline source/index.txt:22]: Issue: the use of "it" here is unclear as you also mentioned a few other nouns in the previous sentence.

```suggestion
Download the driver using `NuGet <https://www.nuget.org/>`__, or set up a runnable
```

Comment (rustagir) [inline source/index.txt:22]: Question: would it be helpful for the NuGet link to bring users to the specific page for the MongoDB Driver?

File: source/index.txt [hunk 6]
```diff
@@ -13,5 +13,59 @@ MongoDB C# Driver
...
+-------------
+
+Learn how to report bugs, contribute to the driver, and find
+additional resources for asking questions in the :ref:`csharp-issues-help` section.
+
+FAQ
+---
+
+For answers to commonly asked questions about the {+driver-long+}, see :ref:`csharp-faq`
+section.
```
Comment (rustagir) [inline source/index.txt:71]:
```suggestion
For answers to commonly asked questions about the {+driver-long+}, see the :ref:`csharp-faq`
section.
```

File: source/usage-examples.txt [hunk 7]
```diff
@@ -16,4 +20,43 @@ TODO: Write usage examples page
...
+   /usage-examples/deleteMany
+
+Overview
+--------
+
+Usage examples provide convenient starting points for popular MongoDB
+operations. Each example provides the following information:
+
+- Code snippets showing methods to perform the operation synchronously and
+  asynchronously
```
Comment (rustagir) [inline source/usage-examples.txt:32]:
```suggestion
- A code snippet that shows how to perform the operation in synchronous and asynchronous frameworks
```

File: source/usage-examples.txt [hunk 8]
```diff
@@ -16,4 +20,43 @@ TODO: Write usage examples page
...
+
+- A fully runnable console application
```
Comment (rustagir) [inline source/usage-examples.txt:34]: Issue: the phrasing of this list item doesn't flow well with the introductory text

```suggestion
- A link to a fully runnable console application using the operation
```

File: source/usage-examples/deleteMany.txt [hunk 9]
```diff
@@ -72,10 +72,8 @@ Running either of the preceding full examples prints the following results:
    Deleted documents: 102
    Resetting sample data...done.
 
-More Information
-----------------
-
-.. TODO: Add links to references once pages are done
+Additional Information
+----------------------
```
Comment (rustagir) [inline source/usage-examples/deleteMany.txt:76]: Suggestion: link to Delete fundamentals page

File: source/usage-examples/deleteOne.txt [hunk 10]
```diff
@@ -84,8 +84,6 @@ Running either of the preceding full examples prints the following results:
 Additional Information
 ----------------------
```
Comment (rustagir) [inline source/usage-examples/deleteOne.txt:85]: Suggestion: link to Delete fundamentals page

File: source/usage-examples/findMany.txt [hunk 11]
```diff
@@ -170,10 +170,10 @@ Running the preceding full examples prints the following results:
...
 ----------------------
 
-.. TODO: Add links to references once pages are done
-
 To learn more about using builders, see :ref:`csharp-builders`.
 
+To learn how to find documents using LINQ, see :ref:`csharp-linq`.
+
 API Documentation
 ~~~~~~~~~~~~~~~~~
```
Comment (rustagir) [inline source/usage-examples/findMany.txt:178]: Issue: in all of the other usage examples, the API documentation is a top level header, not nested under Additional Information. 

Suggestion: either make this header a top level header or change all the other examples to match this one.

File: source/usage-examples/findMany.txt [hunk 12]
```diff
@@ -170,10 +170,10 @@ Running the preceding full examples prints the following results:
 Additional Information
 ----------------------
```
Comment (rustagir) [inline source/usage-examples/findMany.txt:171]: Suggestion: link to Retrieve Data fundamentals page

File: source/usage-examples/findOne.txt [hunk 13]
```diff
@@ -120,10 +120,11 @@ Running any of the preceding full examples prints the following results:
 
 Additional Information
 ----------------------
```
Comment (rustagir) [inline source/usage-examples/findOne.txt:122]: Suggestion: link to Retrieve Data fundamentals page

File: source/usage-examples/replaceOne.txt [hunk 14]
```diff
@@ -78,8 +78,6 @@ Running either of the preceding full examples prints the following results:
 Additional Information
 ----------------------
```
Comment (rustagir) [inline source/usage-examples/replaceOne.txt:79]: Suggestion: link to Change Documents fundamentals page

File: source/usage-examples/updateMany.txt [hunk 15]
```diff
@@ -77,8 +77,6 @@ Running either of the preceding full examples prints the following results:
 More Information
 ----------------
```
Comment (rustagir) [inline source/usage-examples/updateMany.txt:78]: Suggestion: link to Change Documents fundamentals page

File: source/usage-examples/updateOne.txt [hunk 16]
```diff
@@ -89,7 +89,6 @@ writes the following to the console:
 
 More Information
 ----------------
```
Comment (rustagir) [inline source/usage-examples/updateOne.txt:91]: Suggestion: link to Change Documents fundamentals page

**Review by rustagir — CHANGES_REQUESTED** [review]
Looks good, there are some missing references and some pages need to be updated and added as TOC landing pages. Also -- I can't figure out how to search through all the files in your branch but did you check that there are not still any outstanding TODOs?

File: source/index.txt (hunk 5 above)
Comment (jordan-smith721) [reply source/index.txt]: Good call! Linked to the MongoDB driver instead

**Comment by jordan-smith721** [comment]
> Looks good, there are some missing references and some pages need to be updated and added as TOC landing pages. Also -- I can't figure out how to search through all the files in your branch but did you check that there are not still any outstanding TODOs?

There are still some outstanding TODOs in the repo but they are ones for pages that aren't written yet (and all are in comments so they won't show up)

**Review by rustagir — APPROVED** [review]
LGTM! Looks great

**Pull request merged**